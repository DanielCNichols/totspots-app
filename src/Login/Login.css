
.login {
  width: 100%;
  margin: 100px auto 0 auto;
  text-align: center;
}

.login label {
  text-align: center;
  display: block;
  margin: 1.5em;
}

.login_text {
  display: block;
  width: 100%;
  margin: 5px auto;
  padding: 5px 10px;
  height: 2.5rem;
  min-width: 125px;
  border: 1px solid black;
  border-left: 3px solid;
  border-radius: 5px; 
  transition: border-color .5s ease-out;
}

.login label {
  text-align: left;
  width: 80%;
  margin: 20px auto;
}



.login button {
  display: block;
  min-width: 180px;
  height: 40px;
  margin: 20px auto;
  border-radius: 5px;
  cursor: pointer;
  background-color: transparent;
  font-size: 1.1em;
  border-color: #064789;

}

@media screen and (min-width: 450px) {
  .login {
    width: 60%;
    margin: 100px auto;
  }

  .login button {
    width: 40%;
  }
}

@media screen and (min-width: 800px) {
  .login {
    width: 30%;
    margin-top: 100px;
  }

  .login button {
    width: 20%;
  }
}