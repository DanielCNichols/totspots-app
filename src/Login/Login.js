import React, {Component} from 'react';
import TokenService from '../services/TokenService'
import AuthService from '../services/AuthService'


export default class Login extends Component {
  static defaultProps = {
    location: {},
    history: {
      push: () => {},
    },
  }

  state = {error: null}
  
  handleLogin = () => {
    const {location, history} = this.props
    const destination = (location.state || {}).from || '/'
    history.push(destination)
  }

  handleSubmitJwtAuth = ev => {
    console.log('submitting')
    ev.preventDefault() 
    this.setState({error: null})
    const {user_name, password} = ev.target
    console.log(`${user_name}, ${password}}`)

    AuthService.postLogin({
      user_name: user_name.value,
      password: password.value,
    })
    .then(res => {
      user_name.value = ''
      password.value = ''
      TokenService.saveAuthToken(res.authToken)
      this.handleLogin()
    })
    .catch(res => {
      this.setState({error:res.error})
    })
  }


  render() {
    return (
      <section>
      <header>
        <h2>Sign in</h2>
      </header>
      <form action="" onSubmit={this.handleSubmitAuth}>
        <fieldset>
          <label for="user_name">Username
            <input type="text" name="user_name" id="user_name" placeholder="jonDoe82"/>
          </label>
          <label for="password">Password
            <input type="text" name="password" id="password"/>
          </label>
        </fieldset>
      <button>Log in</button>
        <p>Forgot Password?</p>
      </form>
    </section>
    )
  }
}