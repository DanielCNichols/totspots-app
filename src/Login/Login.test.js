import React from 'react';
import ReactDOM from 'react-dom'
import Login from './Login.js'
import renderer from 'react-test-renderer'
import {MemoryRouter} from 'react-router-dom'


it('renders Login without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<Login />, div);
  ReactDOM.unmountComponentAtNode(div)
})


it('renders Login as expected', () => {
  const tree = renderer
  .create(
  <MemoryRouter>
    <Login name="Login"/>
  </MemoryRouter>
  )
  .toJSON();
  expect(tree).toMatchSnapshot();
})
