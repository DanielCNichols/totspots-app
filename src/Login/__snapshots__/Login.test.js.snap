// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Login as expected 1`] = `
<section
  className="login"
>
  <header>
    <h2>
      Sign in
    </h2>
  </header>
  <form
    className="login_form"
    onSubmit={[Function]}
  >
    <fieldset>
      <label
        htmlFor="user_name"
      >
        Username
        <input
          className="login_text"
          id="username"
          name="username"
          placeholder="jonDoe82"
          type="text"
        />
      </label>
      <label
        htmlFor="password"
      >
        Password
        <input
          className="login_text"
          id="password"
          name="password"
          type="text"
        />
      </label>
    </fieldset>
    <button>
      Log in
    </button>
  </form>
</section>
`;
